compiler-rt/test/lit.common.cfg.py

```diff
@@ -965,6 +965,16 @@ def is_windows_lto_supported():
 else:
     config.available_features.add("memprof-shadow-scale-3")
 
+def target_page_size():
+    try:
+        proc = subprocess.Popen(f"{emulator or ""} python3", shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
```

Comment:
Do we really need `shell`? And, can we use `check_output` instead?

```
return int(subprocess.check_output(([emulator] if emulator else []) + ['python3', '-c', 'import os; print(os.sysconf("SC_PAGESIZE"))']))
```

Reply:
I think there's some flexibility in what `emulator` is allowed to be. It could be something like `ssh foo` which means that the shell needs to split `ssh` and `foo`. And then in that case the shell quoting rules end up messing up arguments with spaces (the arguments to ssh get joined together with `" "` and passed as a single argument to `sh -c` so part of the Python script is interpreted by the shell as a shell command). If we want something that works with and without ssh it seemed the simplest way was to pass the script via stdin.